src/main/java/frc/robot/subsystems/FeederSubsystem.java

```diff
@@ -0,0 +1,130 @@
+
+    //IR sensor that monitors the indexer
+    private static AnalogInput indexerIrSensor;
+
+    //How much to shift the feeder wheel when incrementing
+    public static FluidConstant<Double> INCREMENT_TICKS = new FluidConstant<>("IncrementTicks", 0.0)
+                .registerToTable(Config.constantsTable);
+    //Max distance at which the robot knows a ball is at the indexer
+    public static FluidConstant<Integer> IR_MAX_DISTANCE = new FluidConstant<>("IrMaxDistance", 0)
+                .registerToTable(Config.constantsTable);
+    public static FluidConstant<Double> FEEDERSUBSYSTEM_P = new FluidConstant<>("FeederSubsystemP", 0.0)
+                .registerToTable(Config.constantsTable);
+    public static FluidConstant<Double> FEEDERSUBSYSTEM_I = new FluidConstant<>("FeederSubsystemI", 0.0)
+                .registerToTable(Config.constantsTable);
+    public static FluidConstant<Double> FEEDERSUBSYSTEM_D = new FluidConstant<>("FeederSubsystemD", 0.0)
+                .registerToTable(Config.constantsTable);
+    public static FluidConstant<Double> FEEDERSUBSYSTEM_F = new FluidConstant<>("FeederSubsystemF", 0.0)
+                .registerToTable(Config.constantsTable);
+    //Highest speed the motor could reach
+    public static FluidConstant<Double> FEEDERSUBSYSTEM_PEAK_OUTPUT = new FluidConstant<>("FeederSubsystemPeakOutput", 0.0) 
+                .registerToTable(Config.constantsTable);
+
+    private final int kTimeoutMs = 1000;
+
+    private final int kPIDLoopIdx = 0;
+
+    private FeederSubsystem(){
+
+        //Initialize the talon
+        feederTalon = new WPI_TalonSRX(Config.FEEDERSUBSYSTEM_TALON);
+
+        //Initialize the IR sensor
+        indexerIrSensor = new AnalogInput(Config.FEEDERSUBSYSTEM_IR_SENSOR);
+
+        //Configure the talon
+        feederTalon.configFactoryDefault();
+        feederTalon.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, kPIDLoopIdx, kTimeoutMs);
+        feederTalon.setSensorPhase(true);
+        feederTalon.configNominalOutputForward(0, kTimeoutMs);
+        feederTalon.configNominalOutputReverse(0, kTimeoutMs);
+        feederTalon.configPeakOutputForward(FEEDERSUBSYSTEM_PEAK_OUTPUT.get(), kTimeoutMs);
+        feederTalon.configPeakOutputReverse(-FEEDERSUBSYSTEM_PEAK_OUTPUT.get(), kTimeoutMs);
+        feederTalon.configAllowableClosedloopError(0, 0, kTimeoutMs);
+        feederTalon.config_kF(kPIDLoopIdx, FEEDERSUBSYSTEM_F.get(), kTimeoutMs);
+        feederTalon.config_kP(kPIDLoopIdx, FEEDERSUBSYSTEM_P.get(), kTimeoutMs);
+        feederTalon.config_kI(kPIDLoopIdx, FEEDERSUBSYSTEM_I.get(), kTimeoutMs);
+        feederTalon.config_kD(kPIDLoopIdx, FEEDERSUBSYSTEM_D.get(), kTimeoutMs);
+        int absolutePosition = feederTalon.getSensorCollection().getPulseWidthPosition();
+        feederTalon.setSelectedSensorPosition(absolutePosition, kPIDLoopIdx, kTimeoutMs);
+        feederTalon.setSelectedSensorPosition(0, 0, 10);
+
```

Comment:
Some of these calls return errorCodes - most critically the call to configSelectedFeedbackSensor(). We need to catch the error code and do something about anything abnormal, such as if the encoder isn't found by the Talon.

Nathan has written a cool conditions framework that lets you set a condition called "encoderFound" or "encoderHealthy" or whatever you want to call it. The condition framework allows the subsystem to determine if it should run or not based on every condition being met. You could use that to essentially flag the subsystem as being unable to run if the errorCode is not happy. We should also log the errors.